lib/ansible/plugins/connection/ssh.py

```diff
@@ -550,8 +550,14 @@ def _run(self, cmd, in_data, sudoable=True):
         if p.returncode != 0 and controlpersisterror:
             raise AnsibleError('using -c ssh on certain older ssh versions may not support ControlPersist, set ANSIBLE_SSH_ARGS="" (or ssh_args in [ssh_connection] section of the config file) before running again')
 
+        # If we find a broken pipe because of ControlPersist timeout expiring (see #16731),
+        # we raise an internal error so that we can retry a connection.
+        controlpersist_broken_pipe = 'mux_client_hello_exchange: write packet: Broken pipe' in stderr
+        if p.returncode == 255 and controlpersist_broken_pipe:
+            raise AnsibleInternalError('SSH Error: data could not be sent because of ControlPersist broken pipe.')
```

Comment:
Since this is a generic exception, I think we need a specific error code or something to ensure that if someone else tosses this down the line for another reason, that the associated retry doesn't get triggered.